spec-examples/src/ntru_prime/ntru_prime.rs

```diff
@@ -0,0 +1,147 @@
+use hacspec::prelude::*;
+
+/// Struct to decide NtruVersion
+pub struct NtruVersion {
+    pub p: usize,
+    pub q: i128,
+    pub w: i128,
+    pub irr: Seq<i128>,
+}
+pub fn set_irr(p: usize) -> Seq<i128> {
+    let mut irr: Seq<i128> = Seq::new(p + 1);
+    irr[0] = -1 as i128;
+    irr[1] = -1 as i128;
+    irr[p] = 1 as i128;
+    irr
+}
+#[macro_export]
+macro_rules! ntru_v {
+    ($t:expr) => {{
+        if $t == 0 {
+            NtruVersion {
+                p: 761,
+                q: 4591,
+                w: 286,
+                irr: set_irr(761),
+            }
+        } else if $t == 1 {
+            NtruVersion {
+                p: 653,
+                q: 4621,
+                w: 288,
+                irr: set_irr(653),
+            }
+        } else {
+            NtruVersion {
+                p: 857,
+                q: 5167,
+                w: 322,
+                irr: set_irr(857),
+            }
+        }
+    }};
+}
+
+///This function creates a random polynom with w many -1 or 1 and with the highes degree of h_deg.
+pub fn create_rand_poly(w: i128, h_deg: usize) -> Seq<i128> {
```

Comment:
This has to be pulled out (no randomness in the spec).
Can you just pull all the code that requires randomness (`key_gen`) out into a separate module. Then we can decide what to do with it.